### PR TITLE
Windows: define WIN32_LEAN_AND_MEAN before including windows.h.

### PR DESCRIPTION
**Pull request by vtorri**
Fix issue #2225. It will also speed up compilation on Windows.

### CODE REVIEW

**Review by hermet — APPROVED** [review]

File: src/tools/svg2tvg/svg2tvg.cpp [hunk 1]
```diff
@@ -25,7 +25,11 @@
 #include <thread>
 #include <thorvg.h>
 #ifdef _WIN32
+    #ifndef WIN32_LEAN_AND_MEAN
+        #define WIN32_LEAN_AND_MEAN
```
Comment (rinechran) [inline src/tools/svg2tvg/svg2tvg.cpp:29]: same

File: src/tools/svg2png/svg2png.cpp [hunk 2]
```diff
@@ -26,7 +26,11 @@
 #include <vector>
 #include "lodepng.h"
 #ifdef _WIN32
+    #ifndef WIN32_LEAN_AND_MEAN
+        #define WIN32_LEAN_AND_MEAN
```
Comment (rinechran) [inline src/tools/svg2png/svg2png.cpp:30]: same

File: src/tools/lottie2gif/lottie2gif.cpp [hunk 3]
```diff
@@ -25,7 +25,11 @@
 #include <vector>
 #include <thorvg.h>
 #ifdef _WIN32
+    #ifndef WIN32_LEAN_AND_MEAN
+        #define WIN32_LEAN_AND_MEAN
```
Comment (rinechran) [inline src/tools/lottie2gif/lottie2gif.cpp:29]: same

File: src/renderer/wg_engine/tvgWgRenderer.cpp [hunk 4]
```diff
@@ -24,7 +24,11 @@
 
 #ifdef _WIN32
 // TODO: cross-platform realization
-#include <windows.h>
+    #ifndef WIN32_LEAN_AND_MEAN
```
Comment (rinechran) [inline src/renderer/wg_engine/tvgWgRenderer.cpp:27]: same

**Review by SergeyLebedkin — APPROVED** [review]

**Review by rinechran — APPROVED** [review]

File: src/loaders/ttf/tvgTtfLoader.cpp [hunk 5]
```diff
@@ -24,7 +24,11 @@
 #include "tvgTtfLoader.h"
 
 #if defined(_WIN32) && (WINAPI_FAMILY == WINAPI_FAMILY_DESKTOP_APP)
+    #ifndef WIN32_LEAN_AND_MEAN
+        #define WIN32_LEAN_AND_MEAN
+    #endif
     #include <windows.h>
+    #undef WIN32_LEAN_AND_MEAN
```
[the remark this answers is not in the document]
Comment (rinechran) [reply src/loaders/ttf/tvgTtfLoader.cpp:31]: Oh, I got it

File: src/tools/lottie2gif/lottie2gif.cpp [hunk 6]
```diff
@@ -25,7 +25,11 @@
 #include <vector>
 #include <thorvg.h>
 #ifdef _WIN32
+    #ifndef WIN32_LEAN_AND_MEAN
```
Comment (hermet) [inline src/tools/lottie2gif/lottie2gif.cpp:28]: In tools, these are just optional, not mandatory.
probably, we can remove them add them in meson 
or at least just single line of  `#define WIN32_LEAN_AND_MEAN` without any ancillary code.

File: src/loaders/ttf/tvgTtfLoader.cpp (hunk 5 above)
Comment (hermet) [inline src/loaders/ttf/tvgTtfLoader.cpp:31]: If thorvg undef it, then I think thorvg could include it in config.h for applying it globally.

**Review by hermet — COMMENTED** [review]
reviewed again.

**Comment by hermet** [comment]
https://github.com/thorvg/thorvg/pull/2238

**Pull request closed without merging**